infrastructure-playbooks/rolling_update.yml

```diff
@@ -588,6 +588,49 @@
         - containerized_deployment
 
 
+- name: upgrade ceph iscsi gateway node
+
+  vars:
+    upgrade_ceph_packages: True
+
+  hosts:
+    - "{{ iscsi_gw_group_name|default('iscsi-gws') }}"
+
+  serial: 1
+  become: True
+
+  pre_tasks:
+    # failed_when: false is here so that if we upgrade
+    # from a version of ceph that does not have iscsi gws
+    # then this task will not fail
+    - name: stop rbd-target-gw
+      systemd:
+        name: rbd-target-gw
+        state: stopped
+        enabled: yes
+      failed_when: false
+      when:
+        - not containerized_deployment
+
+  roles:
+    - ceph-defaults
+    - { role: ceph-common, when: not containerized_deployment }
```

Comment:
We might need to skip this role this way

`ceph_release_num[ceph_release] >= ceph_release_num.luminous`

otherwise, it will install ceph packages for targeted release during upgrade, then, you will satisfy the conditions below (L621-L622).

Not sure. To be double checked.